Result/Result.swift

```diff
@@ -31,6 +31,15 @@ public enum Result<T, Error: ErrorType>: ResultType, CustomStringConvertible, Cu
 		}
 	}
 	
+	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws, non-autoclosure version
+    	public init?(_ f: () throws -> T?) {
+        	do {
+            		guard let result = try f() else { return nil }
```

Comment:
I agree with @NachoSoto. This should be defined in terms of `T` (and probably marked as `@noescape`). See the [materialize implementations](https://github.com/frogcjn/Result/blob/patch-1/Result/Result.swift#L158-L168) for reference.